### ISSUE TITLE
Property erroneously flagged as unused in alpha 6

### ISSUE THREAD

**Issue by strindberg** [detekt/detekt#9575]
## Expected Behavior

A property used within a class should not be flagged as unused.
## Observed Behavior

A property used within a class is flagged as unused.
## Steps to Reproduce

With this code, `username`and `password` are flagged as unused ([UnusedPrivateProperty]) even though they are used within the class:
```kotlin
import java.net.Authenticator
import java.net.PasswordAuthentication
import java.net.http.HttpClient

class EmailClient(
    private val username: String,
    private val password: String,
) {

    val client: HttpClient =
        HttpClient
            .newBuilder()
            .authenticator(object : Authenticator() {
                override fun getPasswordAuthentication(): PasswordAuthentication = PasswordAuthentication(username, password.toCharArray())
            })
            .build()

}
```

This does not happen in version 2.0.0-alpha.5.

## Your Environment

* Version of detekt used: 2.0.0-alpha.6

**Comment by BraisGabin** [comment]
Dupe of #9571. They are unused properties (but used parameters)
